athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```
@@ -0,0 +1,385 @@
+     * 1. An Apache Arrow Schema object describing the table's columns, types, and descriptions.
+     * 2. A Set<String> of partition column names (or empty if the table isn't partitioned).
+     * 3. A TableName object confirming the schema and table name the response is for.
+     * 4. A catalog name corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public GetTableResponse doGetTable(BlockAllocator allocator, GetTableRequest request) throws SQLException {
+        logger.info("doGetTable: " + request.getTableName());
+
+        SchemaBuilder tableSchemaBuilder = SchemaBuilder.newBuilder();
+        Set<String> partitionCols = new HashSet<>();
+
+        try {
+            Connection client = getOrCreateConn(request);
+            DatabaseMetaData dbMetadata = client.getMetaData();
+            TableName name = request.getTableName();
+            ResultSet definition = dbMetadata.getColumns(null,name.getSchemaName(),name.getTableName(), null);
+            while(definition.next()) {
+
+                //TODO: this needs to change as per the schema of the table in Vertica; Need to add other data types here!!
+                //If Integer
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("Integer")) {
+                    tableSchemaBuilder.addIntField(definition.getString("COLUMN_NAME"));
+                }
+                //If VARCHAR
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("Varchar")) {
+                    tableSchemaBuilder.addStringField(definition.getString("COLUMN_NAME"));
+                }
+                //If Numeric
+                if (definition.getString("TYPE_NAME").equalsIgnoreCase("NUMERIC")) {
+                    tableSchemaBuilder.addDecimalField(definition.getString("COLUMN_NAME"), 10, 2);
+                }
+            }
+
+        }
+        catch (SQLException e) {
+           logger.info("SQL Exception in doGetTable : "+ e.getMessage());
+        }
```

swallowing exceptions is bad practice. In this case you should rethrow the exception...perhaps as 

throw new RuntimeException("SQL Exception in doGetTable." + e.getMessage, e);

This way Athena will see the failure and show it to the customer. As the code currently stands, Athena may not know anyhting went wrong and may throw a weird (not helpful) error to the customer.

---

thank you. replaced with proper exception handling